src/components/Gradebook/index.jsx

```diff
@@ -711,9 +764,17 @@ Gradebook.defaultProps = {
 Gradebook.propTypes = {
   areGradesFrozen: PropTypes.bool,
   assignmentTypes: PropTypes.arrayOf(PropTypes.string),
+  assignmentFilterOptions: PropTypes.arrayOf(PropTypes.shape({
```

Comment:
Only if this component depends on those values being set (which it doesn't; it passes them along if they're there but it doesn't depend on them being set).

Ultimately this question is about how stringently we want to validate our prop shape. Generally I tend to be pretty loose about deeply nested prop types, since I find they often change.